test/integration/associations/has-many.test.js

```diff
@@ -1494,7 +1494,7 @@ describe(Support.getTestDialectTeaser('HasMany'), () => {
             include: [
               { model: Task, separate: true, include: [{ model: Job, separate: true }] }
             ],
-            group: [['userId']]
+            group: ['userId']
```

Comment:
This just became a breaking change :D

Reply:
Not really, the `group` here never worked right since the `intersect` was added which would've removed it.

Reply:
Sounds reasonable to me, this feels like it was broken before.. So IMO we could also go ahead and and push this to v5

Reply:
@SimonSchick 

```sql
Executing (default): SELECT count("user"."userId") AS "count" FROM "users" AS "user" GROUP BY "user"."userId";
Executing (default): SELECT "user"."userId" FROM "users" AS "user" GROUP BY "user"."userId";
Executing (default): SELECT "task"."taskId", "task"."title", "task"."createdAt", "task"."updatedAt", "task"."userId" FROM "tasks" AS "task" WHERE "task"."userId" IN (1);
Executing (default): SELECT "jobId", "title", "createdAt", "updatedAt", "taskId" FROM "jobs" AS "job" WHERE "job"."taskId" IN (1, 2);
```

Seems like `group` is working properly here. You can see `GROUP BY userId` in first two queries.

We can make this change non-breaking by taking care of arrays like `[[item]]`

Reply:
What if you remove the `group` in this case entirely?
What happened to me is that the same query was generated eg. `the double nested `group` had no effect.